cli/src/parse.rs

```diff
@@ -393,6 +526,177 @@ pub fn parse_file_at_path(parser: &mut Parser, opts: &ParseFileOptions) -> Resul
     })
 }
 
+const fn escape_invisible(c: char) -> Option<&'static str> {
+    Some(match c {
+        '\n' => "\\n",
+        '\r' => "\\r",
+        '\t' => "\\t",
+        '\0' => "\\0",
+        '\\' => "\\\\",
+        '\x0b' => "\\v",
+        '\x0c' => "\\f",
+        _ => return None,
+    })
+}
+
+fn render_node_text(source: &str) -> String {
+    let mut escaped = String::new();
+    escaped.reserve(source.len());
+
+    for c in source.chars() {
+        if let Some(esc) = escape_invisible(c) {
+            escaped.push_str(esc);
+        } else {
+            escaped.push(c);
+        }
+    }
+
+    escaped
```

Comment:
A couple of suggestions in one, you don't need to call reserve after creating a new `String`, `with_capacity` is a shorthand version of that, and we can also just leverage `fold` since we're accumulating into the String and nothing else.

```suggestion
    source
        .chars()
        .fold(String::with_capacity(source.len()), |mut acc, c| {
            if let Some(esc) = escape_invisible(c) {
                acc.push_str(esc);
            } else {
                acc.push(c);
            }
            acc
        })
```

Reply:
Ah, that's much nicer